core/templates/components/common-layout-directives/common-elements/sharing-links.component.ts

```diff
@@ -0,0 +1,111 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the Social Sharing Links.
+ */
+const constants = require('constants.ts');
+
+import { Component, Input, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { HtmlEscaperService } from 'services/html-escaper.service';
+import { ExplorationEmbedButtonModalComponent } from
+  'components/button-directives/exploration-embed-button-modal.component';
+
+@Component({
+  selector: 'sharing-links',
+  templateUrl: './sharing-links.component.html',
+  styleUrls: []
+})
+export class SharingLinksComponent implements OnInit {
+  @Input() layoutType: string;
+  @Input() layoutAlignType: string;
+  @Input() shareType: ShareType;
+  @Input() explorationId: string;
+  @Input() collectionId: string;
+  @Input() smallFont: boolean;
+  classroomUrl: string;
+  activityId: string;
+  serverName: string;
+  escapedTwitterText: string;
+
+  constructor(
+    private nbgModal: NgbModal,
+    private urlInterpolationService: UrlInterpolationService,
+    private siteAnalyticsService: SiteAnalyticsService,
+    private htmlEscaperService: HtmlEscaperService) {}
+
+  ngOnInit() {
+    if (this.shareType === 'exploration') {
+      this.activityId = this.explorationId;
+    } else if (this.shareType === 'collection') {
+      this.activityId = this.collectionId;
+    } else {
+      throw new Error(
+        'SharingLinks directive can only be used either in the' +
+        'collection player or the exploration player');
+    }
+
+    this.serverName = (
+      window.location.protocol + '//' + window.location.host);
+
+    this.escapedTwitterText = (
+      this.htmlEscaperService.unescapedStrToEscapedStr(
+        constants.DEFAULT_TWITTER_SHARE_MESSAGE_EDITOR));
+
+    this.classroomUrl = this.urlInterpolationService.getStaticImageUrl(
+      '/general/classroom.png');
+  }
+
+  getUrl(network: SharingPlatform) {
+    if (network === 'facebook') {
+      return `https://www.facebook.com/sharer/sharer.php?sdk=joey&u=${this.serverName}/${this.shareType}/${this.activityId}&display=popup&ref=plugin&src=share_button`;
+    }
+
+    if (network === 'twitter') {
+      return `https://twitter.com/share?text=${this.escapedTwitterText}&url=${this.serverName}/${this.shareType}/${this.activityId}`;
+    }
+
+    if (network === 'classroom') {
+      return `https://classroom.google.com/share?url=${this.serverName}/${this.shareType}/${this.activityId}`;
+    }
+  }
+
+  showEmbedExplorationModal() {
+    const modelRef = this.nbgModal.open(
```

Comment:
Could you add steps for testing this part locally?

Reply:
Sure. To test my changes, follow these steps:

1> Create and exploration and publish it. As soon as you publish you are greeted with a "share" screen. At the end of the modal, you will find four icons. That is the `sharing-links` component. Click on the fourth icon (the icon contains '<>' inside) to open the modal migrated here.

2> After you have created an exploration, go to creator-dashboard and you will find your explorations in small cards. At the very bottom of the exploration card, you will find the same four icons (they will look smaller). Click on the fourth icon to open the modal migrated here.

3> You can also play the exploration. When you are playing the exploration, you will find a small (i) icon in the navbar (on the left side). Click on it and you will find the four icons again. Click on the fourth icon to open the modal migrated here.

There is also a fourth (final) place where this directive is used. But I think these three should be enough. :)